src/FRP/Event.purs

```diff
@@ -42,6 +44,18 @@ foreign import never :: forall a. Event a
 instance functorEvent :: Functor Event where
   map = mapImpl
 
+instance filterableEvent :: Filterable Event where
+  filter = filter
+
+  filterMap f = unsafePartial (map fromJust <<< filter isJust <<< map f)
+
+  partition p xs = { yes: filter p xs, no: filter (not <<< p) xs }
+
+  partitionMap f xs = let ys = f <$> xs in
```

Comment:
I think we can define this a little more succinctly in terms of `filterMap`.

Reply:
Yeah, I think I was trying to be clever and save a call, but it actually just ends up being a mess. Will do.